src/spaces/graph.jl

```diff
@@ -91,11 +91,7 @@ nodes(model::ABM{<:AbstractAgent,<:GraphSpace}) = 1:nv(model)
 #######################################################################################
 function space_neighbors(pos::Int, model::ABM{A,<:GraphSpace}, args...; kwargs...) where {A}
     nn = node_neighbors(pos, model, args...; kwargs...)
-    # TODO: Use flatten here or something for performance?
-    # `model.space.s[nn]...` allocates, because it creates a new array!
-    # Also `vcat` allocates a second time
-    # We include the current node in the search since we are searching over space
-    vcat(model.space.s[pos], model.space.s[nn]...)
+    Iterators.flatten(model.space.s[i] for i in Iterators.flatten((pos,nn)))
```

Comment:
perhaps it is more performant to do `push!(nn, pos)` instead of `Iterators.flatten((pos, nn))`?

Reply:
You have to `copy(nn)` to do this, leading to more time and allocations.

Reply:
`node_neighbors` is the result of finally calling `LightGraph.neighbors`, which returns `Vector{Int}`, right?

Reply:
Yes, but the call is by reference, so when you start pushing numbers into `nn` you're adding nodes to the graph and things stop working very quickly.

Reply:
What??? Are you sure that `LightGraph.neighbors` returns a reference? Because calling `LightGraph.neighbors` **allocates**.

Reply:
Haven't gone into the depths of the implementation, but we certainly hit the mutation issue that made us do #281, so yeah&mdash;it's an issue that needs to be considered.